plugins/JSAPIExample/src/JSAPIExample.cpp

```diff
@@ -0,0 +1,250 @@
+//
+//  JSAPIExample.cpp
+//  plugins/JSAPIExample/src
+//
+//  Copyright (c) 2019-2020 humbletim (humbletim@gmail.com)
+//  Copyright (c) 2019 Kalila L. (kasenvr@gmail.com)
+//
+//  Distributed under the Apache License, Version 2.0.
+//  See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
+
+//  Example of prototyping new JS APIs by leveraging the existing plugin system.
+
+#include <QCoreApplication>
+#include <QtCore/QByteArray>
+#include <QtCore/QDir>
+#include <QtCore/QFileInfo>
+#include <QtCore/QJsonObject>
+#include <QtCore/QLoggingCategory>
+#include <QtCore/QThread>
+#include <QtCore/QTimer>
+#include <QtScript/QScriptEngine>
+#include <QtScript/QScriptable>
+
+#include <SettingHelpers.h>  // for ::settingsFilename()
+#include <SharedUtil.h>      // for ::usecTimestampNow()
+#include <shared/ScriptInitializerMixin.h>
+
+// NOTE: replace this with your own namespace when starting a new plugin (to avoid .so/.dll symbol clashes)
+namespace REPLACE_ME_WITH_UNIQUE_NAME {
+
+static constexpr auto JSAPI_SEMANTIC_VERSION = "0.0.1";
+static constexpr auto JSAPI_EXPORT_NAME = "JSAPIExample";
+
+QLoggingCategory logger{ "jsapiexample" };
```

Comment:
```suggestion
QLoggingCategory logger { "jsapiexample" };
```
Existing code / coding standard prefers a space.